modules/apps/account/account-service/src/main/java/com/liferay/account/internal/instance/lifecycle/AddAccountEntryContactListTypesPortalInstanceLifecycleListener.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.account.internal.instance.lifecycle;
+
+import com.liferay.account.constants.AccountListTypeConstants;
+import com.liferay.portal.instance.lifecycle.BasePortalInstanceLifecycleListener;
+import com.liferay.portal.instance.lifecycle.PortalInstanceLifecycleListener;
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.model.ListType;
+import com.liferay.portal.kernel.service.ListTypeLocalService;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+
+import java.util.Map;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Stefano Motta
+ */
+@Component(service = PortalInstanceLifecycleListener.class)
+public class AddAccountEntryContactListTypesPortalInstanceLifecycleListener
+	extends BasePortalInstanceLifecycleListener {
+
+	@Override
+	public void portalInstanceRegistered(Company company) throws Exception {
+		for (Map.Entry<String, String[]> entry :
+				_accountEntryContactListTypes.entrySet()) {
+
+			for (String value : entry.getValue()) {
+				if (!_hasListType(
+						company.getCompanyId(), value, entry.getKey())) {
+
+					_listTypeLocalService.addListType(
+						company.getCompanyId(), value, entry.getKey());
+				}
+			}
+		}
+	}
+
+	private boolean _hasListType(long companyId, String name, String type) {
+		ListType listType = _listTypeLocalService.getListType(
```

Comment:
isn't the throw of the exception breaking the cycle too? shouldn't it just skip in the case the type is not there rather then terminate this process?

Reply:
the _listTypeLocalService.getListType doesn't throw an exception